x-pack/plugins/timelines/public/components/hover_actions/actions/add_to_timeline.tsx

```diff
@@ -41,8 +43,59 @@ export const useGetHandleStartDragToTimeline = ({
   return handleStartDragToTimeline;
 };
 
-export const AddToTimelineButton: React.FC<HoverActionComponentProps> = React.memo(
-  ({ field, onClick, ownFocus, showTooltip = false, value }) => {
+export interface AddToTimelineButtonProps extends HoverActionComponentProps {
+  draggableId?: DraggableId;
+  dataProvider?: DataProvider[] | DataProvider;
+}
+
+const AddToTimelineButton: React.FC<AddToTimelineButtonProps> = React.memo(
+  ({
+    closePopOver,
+    dataProvider,
+    defaultFocusedButtonRef,
+    draggableId,
+    field,
+    keyboardEvent,
+    ownFocus,
+    showTooltip = false,
+    value,
+  }) => {
+    const dispatch = useDispatch();
+    const { addSuccess } = useAppToasts();
+    const startDragToTimeline = useGetHandleStartDragToTimeline({ draggableId, field });
+    const handleStartDragToTimeline = useCallback(() => {
+      if (draggableId != null) {
+        startDragToTimeline();
+      } else if (!isEmpty(dataProvider)) {
+        const addDataProvider = Array.isArray(dataProvider) ? dataProvider : [dataProvider];
+        addDataProvider.forEach((provider) => {
+          if (provider) {
+            dispatch(
+              tGridActions.addProviderToTimeline({
```

Comment:
Thanks for cleaning this up! @angorayc These changes will probably cause conflict with your bug fixes fyi